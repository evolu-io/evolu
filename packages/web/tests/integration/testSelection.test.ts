import { pageUrl } from './helpers/pageUrl';
import { pageDom } from './helpers/pageDom';

beforeEach(async () => {
  await page.goto(pageUrl('testSelection'));
  await page.waitFor(50);
});

test('select', async () => {
  await expect(await pageDom()).toMatchSnapshot();

  await page.click('.select-first-two-letters');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.click('.select-first-two-letters-backward');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.keyboard.down('Shift');
  await page.keyboard.press('ArrowRight');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.click('.select-all');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.click('.select-all-backward');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  // TODO: Once we decide what we want.
  // await page.click('.unselect');
  // await page.waitFor(50);
  // await expect(await pageDom()).toMatchSnapshot();
});

// test('unselect selection', async () => {
//   await page.click('.unselect');
//   await page.waitFor(50);
//   await expect(await pageDom()).toMatchSnapshot();
// });

// test('selection via mousedown', async () => {

// });

// https://github.com/steida/slad/issues/14
export default undefined;
