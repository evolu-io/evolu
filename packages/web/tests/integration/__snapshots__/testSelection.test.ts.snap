// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select 1`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div>
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    style="height:48px;overflow:auto"
  >
    <span
      style="color:#888"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0·
      · ·
      ·{}·
    </span>
    <br />
    <br />
  </pre>
  <div>
    <button
      class="select-first-two-letters"
      type="button"
    >
      ·select first two letters·
    </button>
    <button
      class="select-first-two-letters-backward"
      type="button"
    >
      ·select first two letters backward·
    </button>
    <button
      class="select-all"
      type="button"
    >
      ·select all·
    </button>
    <button
      class="select-all-backward"
      type="button"
    >
      ·select all backward·
    </button>
    <button
      class="unselect"
      type="button"
    >
      ·unselect·
    </button>
  </div>
</div>
`;

exports[`select 2`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div>
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    style="height:48px;overflow:auto"
  >
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,2]}}·
    </span>
    <br />
    <br />
    <span
      style="color:#888"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0·
      · ·
      ·{}·
    </span>
    <br />
    <br />
  </pre>
  <div>
    <button
      class="select-first-two-letters"
      type="button"
    >
      ·select first two letters·
    </button>
    <button
      class="select-first-two-letters-backward"
      type="button"
    >
      ·select first two letters backward·
    </button>
    <button
      class="select-all"
      type="button"
    >
      ·select all·
    </button>
    <button
      class="select-all-backward"
      type="button"
    >
      ·select all backward·
    </button>
    <button
      class="unselect"
      type="button"
    >
      ·unselect·
    </button>
  </div>
</div>
`;

exports[`select 3`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div>
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    style="height:48px;overflow:auto"
  >
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,0]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,2]}}·
    </span>
    <br />
    <br />
    <span
      style="color:#888"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0·
      · ·
      ·{}·
    </span>
    <br />
    <br />
  </pre>
  <div>
    <button
      class="select-first-two-letters"
      type="button"
    >
      ·select first two letters·
    </button>
    <button
      class="select-first-two-letters-backward"
      type="button"
    >
      ·select first two letters backward·
    </button>
    <button
      class="select-all"
      type="button"
    >
      ·select all·
    </button>
    <button
      class="select-all-backward"
      type="button"
    >
      ·select all backward·
    </button>
    <button
      class="unselect"
      type="button"
    >
      ·unselect·
    </button>
  </div>
</div>
`;

exports[`select 4`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div>
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    style="height:48px;overflow:auto"
  >
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,1]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,0]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,2]}}·
    </span>
    <br />
    <br />
    <span
      style="color:#888"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0·
      · ·
      ·{}·
    </span>
    <br />
    <br />
  </pre>
  <div>
    <button
      class="select-first-two-letters"
      type="button"
    >
      ·select first two letters·
    </button>
    <button
      class="select-first-two-letters-backward"
      type="button"
    >
      ·select first two letters backward·
    </button>
    <button
      class="select-all"
      type="button"
    >
      ·select all·
    </button>
    <button
      class="select-all-backward"
      type="button"
    >
      ·select all backward·
    </button>
    <button
      class="unselect"
      type="button"
    >
      ·unselect·
    </button>
  </div>
</div>
`;

exports[`select 5`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div>
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    style="height:48px;overflow:auto"
  >
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·4·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[1,0,9]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,1]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,0]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,2]}}·
    </span>
    <br />
    <br />
    <span
      style="color:#888"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0·
      · ·
      ·{}·
    </span>
    <br />
    <br />
  </pre>
  <div>
    <button
      class="select-first-two-letters"
      type="button"
    >
      ·select first two letters·
    </button>
    <button
      class="select-first-two-letters-backward"
      type="button"
    >
      ·select first two letters backward·
    </button>
    <button
      class="select-all"
      type="button"
    >
      ·select all·
    </button>
    <button
      class="select-all-backward"
      type="button"
    >
      ·select all backward·
    </button>
    <button
      class="unselect"
      type="button"
    >
      ·unselect·
    </button>
  </div>
</div>
`;

exports[`select 6`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div>
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    style="height:48px;overflow:auto"
  >
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·5·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[1,0,9],"focus":[0,0,0]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·4·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[1,0,9]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,1]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,2],"focus":[0,0,0]}}·
    </span>
    <br />
    <br />
    <span
      style="color: rgb(136, 136, 136);"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,2]}}·
    </span>
    <br />
    <br />
    <span
      style="color:#888"
      title="{'children':[{'props':{'style':{'fontSize':'24px'}},'children':['heading']},{'props':{'style':{'fontSize':'16px'}},'children':['paragraph']}]}"
    >
      ·0·
      · ·
      ·{}·
    </span>
    <br />
    <br />
  </pre>
  <div>
    <button
      class="select-first-two-letters"
      type="button"
    >
      ·select first two letters·
    </button>
    <button
      class="select-first-two-letters-backward"
      type="button"
    >
      ·select first two letters backward·
    </button>
    <button
      class="select-all"
      type="button"
    >
      ·select all·
    </button>
    <button
      class="select-all-backward"
      type="button"
    >
      ·select all backward·
    </button>
    <button
      class="unselect"
      type="button"
    >
      ·unselect·
    </button>
  </div>
</div>
`;
